tests/test_compile_kwargs.py

```diff
@@ -0,0 +1,555 @@
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # Test iterable with global routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss=[loss],
+        loss__from_logits=True,  # default is False
+    )
+    est.fit(X, y)
+    assert est.model_.loss[0].from_logits
+
+    # Test iterable with index-based routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss=[loss],
+        loss__from_logits=True,
+        loss__0__from_logits=False,  # should override above
+    )
+    est.fit(X, y)
+    assert est.model_.loss[0].from_logits == False
+
+
+@pytest.mark.parametrize(
+    "loss", (losses_module.BinaryCrossentropy, "BinaryCrossentropy",),
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_loss_routed_params_dict(loss, n_outputs_):
+    """Tests compiling of loss when it is
+    given as an dict of losses
+    mapping to outputs.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # Test dict with global routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss={"out1": loss},
+        loss__from_logits=True,  # default is False
+    )
+    est.fit(X, y)
+    assert est.model_.loss["out1"].from_logits == True
+
+    # Test dict with key-based routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss={"out1": loss},
+        loss__from_logits=True,
+        loss__out1__from_logits=False,  # should override above
+    )
+    est.fit(X, y)
+    assert est.model_.loss["out1"].from_logits == False
+
+
+@pytest.mark.parametrize(
+    "metrics", ["binary_accuracy", metrics_module.BinaryAccuracy]
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_metrics_single_metric_per_output(metrics, n_outputs_):
+    """Metrics without the ("name", metric, "output")
+    syntax should ignore all routed and custom options.
+
+    This tests a single metric per output.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # loss functions for each output and joined show up as metrics
+    metric_idx = 1 + (n_outputs_ if n_outputs_ > 1 else 0)
+    prefix = "out1_" if n_outputs_ > 1 else ""
+
+    expected_name = metrics
+    if metrics != "binary_accuracy":
+        # Test discovery in VSCode fails if TF prints out _any_
+        # warnings during discovery
+        # (which of course it does if you try to instantiate anything)
+        expected_name = metrics().name
+
+    # List of metrics
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=[metrics if not isclass(metrics) else metrics()],
```

Comment:
Suggested change:

``` diff
-         metrics=[metrics if not isclass(metrics) else metrics()],
+         metrics=[metrics if not isinstance(metrics, tf.keras.metrics.Metric) else metrics()],
````

Reply:
Idem other comment as to why I did this in the first place, but also like the other comment this works so glad to change it.